packages/react-frontend/src/routes/Album.jsx

```diff
@@ -20,6 +19,40 @@ async function fetchArtistBySpotifyId(spotifyId) {
   }
 }
 
+// Function to fetch reviews and calculate floored average rating
+async function fetchReviewsAndCalculateRatings() {
+  try {
+    const response = await fetch("http://localhost:8000/reviews");
+    if (!response.ok) {
+      throw new Error("Failed to fetch reviews");
+    }
+    const reviews = await response.json();
+    const albumRatings = {};
+
+    // Calculate the total rating and count for each album
+    reviews.forEach((review) => {
+      const { album_id, rating } = review;
+      if (!albumRatings[album_id]) {
+        albumRatings[album_id] = { totalRating: 0, count: 0 };
+      }
+      albumRatings[album_id].totalRating += rating;
+      albumRatings[album_id].count += 1;
+    });
+
+    // Calculate the floored average rating for each album
+    const averageRatings = {};
+    for (const albumId in albumRatings) {
+      const { totalRating, count } = albumRatings[albumId];
+      averageRatings[albumId] = Math.floor(totalRating / count);
```

Comment:
I like that its floored because we would want, for example, the 3 star filter to show albums reviewed between 3 and 4 stars on average. Its more intuitive than having a 2.9 star album populate under the 3 star filter.

Reply:
Yeah, I had the same thing in mind as Yaesmin, I think that users might get confused if we start rounding up/down the average ratings.